### ISSUE TITLE
tracer context manager ignores parent and trace_id arguments

### ISSUE THREAD

**Issue by lavie** [honeycombio/beeline-python#46]
https://github.com/honeycombio/beeline-python/blob/35acadef447ed99fb1ea20e7f5b87a52bad4c91c/beeline/trace.py#L41

As you can see, the function `__call__` does not make use of these parameters. A linter should have caught this...

As a result, it's impossible to use the context manager to continue a trace across threads, for example.

**Comment by tredman** [comment]
Thanks for the report! I have a fix in the works now and it should be released later today as 2.4.2 after I get it reviewed.

We do run the code through pyflakes and it is usually good at finding things like this, but apparently it doesn't catch unused function arguments because it's sometimes valid to define function arguments that go unused (implementing an interface, for example). I also tried pylint but it also didn't catch it. (it did catch some other issues - yay! so we're going to integrate that into our CI/CD going forward). I did add/update some tests to ensure this behavior works in the future.

**Comment by lavie** [comment]
WRT linters, I do think it's better to turn off such a rule (unused argument) on a line-by-line basis than turn it off completely. It's very rare to intentionally have a function with unused args... (interfaces in python?)...
Anyway, thanks for the speedy response. I would add also that a multi-threaded example of tracing with beeline would be very helpful. Clearly you took the time to make sure it works properly in the face of MT with TLS, so an example of tracing with parallel sub-spans would really shine. I found it tricky to get it to work, having to pass the trace_id and parent span id manually across threads.

**Comment by tredman** [comment]
We do mention it here in our official docs: https://docs.honeycomb.io/beeline/python/#using-traces - see "Threading and Traces". We haven't received any feedback from anyone using it yet though. Are the docs there consistent with your experience? Would appreciate any feedback you have. I'm sure there are things we could do to make it a smoother experience.

This issue will close when I land #47 but if you reply here I'll keep an eye on it. Thanks!

**Comment by lavie** [comment]
Oh I guess I missed that. Will give it a try next week.
Really appreciate the quick fix.

**Comment by lavie** [comment]
@tredman would you agree that the context manager should return a reference to the tracer?

Otherwise we can't do:
```python
with beeline.tracer("name") as trace:
   start_thread_and_pass_trace(trace)
...
```

**Comment by tredman** [comment]
Yes, good catch! I'll open a new issue, thanks.

**Comment by tredman** [comment]
This is fixed in 2.4.3.
